### PR TITLE
Figure.savefig: Support .jpeg as JPEG image extension

### PR DESCRIPTION
**Pull request by seisman**
Support `.jpeg` as file extension. Please note that `gmt psconvert -Tj` always produces images with the `.jpg` extension. It has two downsides:

1. Users may be surprised that they give `.jpeg` but get `.jpg`
2. `show=True` won't work unless we change the codes

I feel it's better to rename the files to have the file extension that users expect. 

Closes https://github.com/GenericMappingTools/pygmt/issues/2689.

### CODE REVIEW

File: pygmt/figure.py [hunk 1]
```diff
@@ -260,9 +260,9 @@ def savefig(
         This method implements a matplotlib-like interface for
         :meth:`pygmt.Figure.psconvert`.
 
-        Supported formats: PNG (``.png``), JPEG (``.jpg``), PDF (``.pdf``),
-        BMP (``.bmp``), TIFF (``.tif``), EPS (``.eps``), and KML (``.kml``).
-        The KML output generates a companion PNG file.
+        Supported formats: PNG (``.png``), JPEG (``.jpg`` or ``.jpeg),
+        PDF (``.pdf``), BMP (``.bmp``), TIFF (``.tif``), EPS (``.eps``), and
+        KML (``.kml``). The KML output generates a companion PNG file.
```
Comment (yvonnefroehlich) [inline pygmt/figure.py:265]:
```suggestion
        Supported formats: PNG (``.png``), JPEG (``.jpg`` or ``.jpeg``),
        PDF (``.pdf``), BMP (``.bmp``), TIFF (``.tif``), EPS (``.eps``), and
        KML (``.kml``). The KML output generates a companion PNG file.
```

**Review by weiji14 — APPROVED** [review]

**Comment by seisman** [comment]
I've revised the codes so that it's much easier to support upper-case extensions (#2683) with very few code changes. Please review again.

File: pygmt/figure.py [hunk 2]
```diff
@@ -304,8 +304,13 @@ def savefig(
             "kml": "g",
         }
 
-        prefix, ext = os.path.splitext(fname)
-        ext = ext[1:]  # Remove the .
+        fname = Path(fname)
+        prefix, suffix = fname.with_suffix("").as_posix(), fname.suffix
+        ext = suffix[1:]
```
Comment (weiji14) [inline pygmt/figure.py:309]:
```suggestion
        ext = suffix[1:]  # Remove the .
```

**Review by weiji14 — APPROVED** [review]
Just adding back the inline comment from `ext = ext[1:]  # Remove the .` before.

**Pull request merged**